lib/jit_preloader/active_record/base.rb

```diff
@@ -131,11 +132,20 @@ def has_many_aggregate(assoc, name, aggregate, field, table_alias_name: nil, def
             end
             group_by = "#{table_reference}.#{aggregate_association.foreign_key}"
 
-            preloaded_data = Hash[association_scope
-              .where(conditions)
-              .group(group_by)
-              .send(aggregate, field)
-            ]
+            if JitPreloader.query_dive_limit
```

Comment:
Would this be clearer if we set the `slices` variable in one place? (Move this up?)